src/benchmark.cc

```diff
@@ -270,14 +269,15 @@ void RunInThread(const benchmark::internal::Benchmark::Instance* b,
   internal::ThreadTimer timer;
   State st(iters, b->arg, thread_id, b->threads, &timer, manager);
   b->benchmark->Run(st);
-  CHECK(st.iterations() == st.max_iterations)
-      << "Benchmark returned before State::KeepRunning() returned false!";
+  CHECK(st.finished()) << "Benchmark returned before State::KeepRunning() / "
+                          "State::KeepRunningBatch() returned false!";
   {
     MutexLock l(manager->GetBenchmarkMutex());
     internal::ThreadManager::Result& results = manager->results;
     results.cpu_time_used += timer.cpu_time_used();
     results.real_time_used += timer.real_time_used();
     results.manual_time_used += timer.manual_time_used();
+    results.iterations += static_cast<int64_t>(st.iterations());
```

Comment:
The cast should be unnecessary here one `results.iterations` is made to have the type `size_t`.